contracts/OperatorWallet.sol

```diff
@@ -0,0 +1,157 @@
+        __Context_init_unchained();
+        __ReentrancyGuard_init_unchained();
+        __ContractWallet_init_unchained(initialOwner, initialOperator);
+    }
+
+    function __ContractWallet_init_unchained(address initialOwner, address initialOperator) internal {
+        require(initialOwner != address(0), "OperatorWallet: owner is a zero address");
+
+        _owner = initialOwner;
+        _operator = initialOperator;
+
+        emit OwnershipTransferred(address(0), initialOwner);
+        emit OperatorTransferred(address(0), initialOperator);
+    }
+
+    function owner() external view returns (address) {
+        return _owner;
+    }
+
+    function operator() external view returns (address) {
+        return _operator;
+    }
+
+    modifier onlyOwner() {
+        require(_owner == _msgSender(), "OperatorWallet: caller is not the owner");
+        _;
+    }
+
+    modifier onlyOperator() {
+        require(_operator == _msgSender(), "OperatorWallet: caller is not the operator");
+        _;
+    }
+
+    function isHotWallet(address account) public view returns (bool){
+        return hotWalletsMap[account] > 0;
+    }
+
+    function transferOwnership(address newOwner) external onlyOwner {
+        require(newOwner != address(0), "OperatorWallet: new owner is a zero address");
+        require(newOwner != _owner, "OperatorWallet: new owner is the same as the current owner");
+
+        emit OwnershipTransferred(_owner, newOwner);
+        _pendingOwner = newOwner;
+    }
+
+    function acceptOwnership() external {
+        require(msg.sender == _pendingOwner, "OperatorWallet: no authorization");
+        emit OwnershipAccepted(_owner, _pendingOwner);
+        _owner = _pendingOwner;
+        _pendingOwner = address(0);
+    }
+
+    function addHotWallet(address hotwallet) external onlyOwner {
+        require(hotwallet != address(0), "OperatorWallet: hotwallet is a zero address");
+        require(hotWalletsMap[hotwallet] == 0, "OperatorWallet: hotwallet already exist");
+        hotWallets.push(hotwallet);
+        hotWalletsMap[hotwallet] = hotWallets.length;
+        numOfHotWallets +=1;
+        emit HotWalletAdded(hotwallet);
+    }
+
+    function removeHotWallet(address hotwallet) external onlyOwner {
+        require(hotwallet != address(0), "OperatorWallet: hotwallet is a zero address");
+        require(hotWalletsMap[hotwallet] > 0, "OperatorWallet: hotwallet do not exist");
+        uint256 idx = hotWalletsMap[hotwallet];
+        if (idx != hotWallets.length) {
+            hotWallets[idx-1] = hotWallets[hotWallets.length-1];
+            hotWalletsMap[hotWallets[idx-1]] = idx;
+        }
+        hotWallets.pop();
+        numOfHotWallets -=1;
+        delete hotWalletsMap[hotwallet];
+        emit HotWalletRemoved(hotwallet);
+    }
+
+    function changeOperator(address newOperator) external onlyOwner {
+        require(newOperator != address(0), "OperatorWallet: operator is a zero address");
+
```

Comment:
do we need to make sure the new operator is not the same as the old one

Reply:
ixed